### PR TITLE
chore: Update deprecations in instrumentation-express/examples

### PR DESCRIPTION
**Pull request by jessitron**
## Which problem is this PR solving?

- The example code for instrumentation-express is out of date.

## Short description of the changes

- This updates the README to make the install work
- the opentelemetry library versions to the latest
- and the tracing code to switch all deprecated imports to their replacement.

### CODE REVIEW

**Comment by trentm** [comment]
Here are a few attempts of my steps to get the express example built and working with a fresh git clone.
I think it might be nice to pick one of these and add some of these exact commands to the README to help newer users get the example working.

# attempt 0

I didn't actually play with and try to get the `npm install --workspaces=false` way. I don't think it is necessarily a bad way to do it, but I think it will end up fighting the workspaces setup a bit.

# attempt 1

Getting example working with a full build of all packages in the repo. This
unfortunately takes a long time.

```bash
# Get a build of the Express example and its dependencies.
git clone git@github.com:open-telemetry/opentelemetry-js-contrib.git
cd opentelemetry-js-contrib
npm ci
npm run compile
cd plugins/node/opentelemetry-instrumentation-express/examples
npm run compile

# Start an Zipkin to use viewing traces.
docker run --name example-zipkin --rm -d -p 9411:9411 openzipkin/zipkin

# Run the express server, configured to export to Zipkin.
npm run zipkin:server

# In a separate terminal, run the client that calls the express server.
npm run zipkin:client
```

Visit [the Zipkin UI](http://localhost:9411/zipkin/?serviceName=example-express-client) to search for spans for the client requests.

When finished then clean up:

```
docker kill example-zipkin
```

# attempt 2 (requires a patch)

I was able to reduce the build time, somewhat by trying to install (via `npm ci`) and compile only in the required subdirs:

```bash
git clone git@github.com:open-telemetry/opentelemetry-js-contrib.git
cd opentelemetry-js-contrib/plugins/node/opentelemetry-instrumentation-express
npm ci
npm run compile
cd examples
npm ci
npm run compile
```

However, that requires this patch to work:

```diff
diff --git a/plugins/node/opentelemetry-instrumentation-express/package.json b/plugins/node/opentelemetry-instrumentation-express/package.json
index 91065940..edfb4477 100644
--- a/plugins/node/opentelemetry-instrumentation-express/package.json
+++ b/plugins/node/opentelemetry-instrumentation-express/package.json
@@ -12,7 +12,7 @@
     "clean": "rimraf build/*",
     "lint": "eslint . --ext .ts",
     "lint:fix": "eslint . --ext .ts --fix",
-    "precompile": "tsc --version && lerna run version:update --scope @opentelemetry/instrumentation-express --include-dependencies",
+    "precompile": "tsc --version && npm run version:update",
     "prewatch": "npm run precompile",
     "version:update": "node ../../../scripts/version-update.js",
     "compile": "tsc -p .",
```

and I'm not yet sure of the implications of doing this in general. Doing this probably should be done on a separate PR that looks at doing it generally for all the packages in this repo.


# attempt 3

This installs all deps at the top-level (`npm ci`), but only runs the compile step in the express instrumentation and examples subdir. This is much faster and works:

```
git clone git@github.com:open-telemetry/opentelemetry-js-contrib.git
cd opentelemetry-js-contrib
npm ci
cd plugins/node/opentelemetry-instrumentation-express
npm run compile
cd examples
npm run compile
```

What do you think about documenting these steps in the README?

**Review by JamieDanielson — APPROVED** [review]
Thanks for this much-needed update!

I understand the frustrations around working with a monorepo. Perhaps we can document both `workspaces=false` for a simple local build and also one of Trent's suggestions around getting the true dependencies working with workspaces. I can add that as a follow-up PR.

**Comment by JamieDanielson** [comment]
Hm looks like the updated merge from main failed some status checks, think we should be able to run `npm ci` to fix it. @jessitron let me know if this is something you can update or if you want any assistance in updating.

**Comment by JamieDanielson** [comment]
Not entirely sure why but I'm unable to push to this PR from my local env. 
I think I fixed this with a merge [commit over here](https://github.com/open-telemetry/opentelemetry-js-contrib/commit/ae10db51601d9b2d90a5f3ebef0577696894404f) if you want to cherry-pick it (ae10db5). Alternatively if anyone else has access to push to this branch (@trentm ) can you help with this? I just did Accept Combination on the merge conflict for the example package.json, and another npm install at the root.

**Comment by trentm** [comment]
> can you help with this?

Hopefully that should do it. I pulled your earlier merge, sync'd the package-lock.json file (`npm install --package-lock-only`), then merged from main again (which had some conflicts), then sync'd package-lock again (not sure why this last small change). The ways of package-lock are mysterious (and not a little frightening).

File: plugins/node/opentelemetry-instrumentation-express/examples/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ shows key aspects of tracing such as
 
 ```sh
 # from this directory
-npm install
+npm install --workspaces=false
```
[the remark this answers is not in the document]
Comment (trentm) [reply plugins/node/opentelemetry-instrumentation-express/examples/README.md:17]: I like the idea of a "setup" (or whatever name) script that just "does all the steps necessary to get this example working, please".

[the remark this answers is not in the document]
Comment (JamieDanielson) [reply plugins/node/opentelemetry-instrumentation-express/examples/README.md:17]: I have the [setup change in this commit ce78071](https://github.com/open-telemetry/opentelemetry-js-contrib/commit/ce78071a7a284136f0749e1266c1f0ee5b396cba) but still having some problems trying to commit to this branch.

[the remark this answers is not in the document]
Comment (JamieDanielson) [reply plugins/node/opentelemetry-instrumentation-express/examples/README.md:17]: @jessitron can you please confirm if this change works for you? If you don't have time I suppose the alternative is that we can remove this change to the README and merge this, and I'll just open a new PR to add this setup script.

[the remark this answers is not in the document]
Comment (trentm) [reply plugins/node/opentelemetry-instrumentation-express/examples/README.md:20]: I pushed it with some git judo that I'm learning as I go:

```
git remote add jamie git@github.com:JamieDanielson/opentelemetry-js-contrib.git
git fetch jamie
git show ce78071a7a284136f0749e1266c1f0ee5b396cba
git cherry-pick ce78071a7a284136f0749e1266c1f0ee5b396cba
git fetch jessitron
git pull --rebase jessitron jessitron/update-example
git push jessitron HEAD
```

File: plugins/node/opentelemetry-instrumentation-express/examples/package.json [hunk 2]
```diff
@@ -26,24 +27,25 @@
...
+    "@opentelemetry/instrumentation-express": "^0.33.3",
     "@opentelemetry/instrumentation-http": "^0.46.0",
-    "@opentelemetry/resources": "^1.0.0",
-    "@opentelemetry/sdk-trace-base": "^1.0.0",
-    "@opentelemetry/sdk-trace-node": "^1.0.0",
-    "@opentelemetry/semantic-conventions": "^1.0.0",
+    "@opentelemetry/resources": "^1.18.1",
+    "@opentelemetry/sdk-trace-base": "^1.18.1",
+    "@opentelemetry/sdk-trace-node": "^1.18.1",
+    "@opentelemetry/semantic-conventions": "^1.18.1",
```
Comment (trentm) [inline plugins/node/opentelemetry-instrumentation-express/examples/package.json:44]: Now these need another update to ^1.20.0 etc, right?

Then I'm not sure if the various examples/package.json files will be included in the deps updates processes.

Comment (trentm) [reply plugins/node/opentelemetry-instrumentation-express/examples/package.json:44]: I was wrong. Those caret version ranges are fine. (They could probably be reduced to lower versions, e.g. `"@opentelemetry/resources": "^1.0.0",`, but I don't think that is required for anything.)

The important thing is that the versions in package-lock.json are good.

**Review by trentm — APPROVED** [review]
Assuming tests pass, I'm happy with this now.

File: plugins/node/opentelemetry-instrumentation-express/examples/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamieDanielson) [reply plugins/node/opentelemetry-instrumentation-express/examples/README.md:20]: For posterity...
I have been using the [`gh pr` GitHub CLI](https://cli.github.com/manual/gh_pr). My default repo was js-contrib, which allows me to run `gh pr checkout 1842` which sets the proper upstream target. After I made a commit, I should have just done `git push`. At least I know [it works ](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1842/commits/b4a6b501578019b8597c87419018888786398af5)now 😅 thanks for the patience!

**Pull request merged**